src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -544,24 +583,27 @@ object IEnumerator.Current
 
             public bool MoveNext()
             {
-                if (_version == _builder._version)
+                ReadOnlyCollectionBuilder<T> localBuilder = _builder;
```

Comment:
It is PR to fix `Quirks with exception behavior of ReadOnlyCollectionBuilder` so I made exception behavior as in `List<T>`.